tests: net: socket: tcp: move to new ztest API

hunterhu3000 opened the pull request:
Move net/socket/tcp tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
CI checks more than "twister -T tests/net/socket/tcp"
Will analyze this problem later

enjiamai commented on tests/net/tcp/src/main.c:
```diff
@@ -1484,23 +1484,4 @@ static void test_server_timeout_out_of_order_data(void)
-void test_main(void)
-{
-	ztest_test_suite(test_tcp_fn,
-			 ztest_unit_test(test_presetup),
```
Try to put the test_presetup in the setup function? It seems like this has to be the first one.

hunterhu3000 commented:
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as some of the test case should be run with specific order.
Thanks to review.

rlubos reviewed (commented):
Please rebase (conflict).

rlubos commented:
@hunterhu3000 Something must've gone wrong during the rebase, the `*.orig` files shouldn't be here.

hunterhu3000 committed:
tests: net: socket: tcp: move to new ztest API

Move net/socket/tcp tests to use new ztest API
TEST=twister -T tests/net/socket/tcp

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
> Please rebase (conflict).

Rebased. Thanks to review

hunterhu3000 commented:
test_z_close_obstructed has to be run at last. If not
```
	zassert_equal(dropped_packets_before, dropped_packets_after,
		      "packets after close");
```
will fail. There will be 1 packet dropped during the test. The reason is unknown yet

enjiamai reviewed (approved):
It looks quite good for me.
